app/questionnaire/navigator.py

```diff
@@ -24,23 +24,55 @@ def evaluate_rule(rule, answer_value):
     return False
 
 
-def evaluate_goto(goto_rule, answers, group_instance):
+def evaluate_goto(goto_rule, metadata, answers, group_instance):
     """
     Determine whether a goto rule will be satisfied based on a given answer
     :param goto_rule:
+    :param metadata
     :param answers:
     :param group_instance:
     :return:
     """
     if 'when' in goto_rule.keys():
-        answer_index = goto_rule['when']['id']
-        filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
-        if len(filtered) == 1:
-            return evaluate_rule(goto_rule, filtered[0]['value'])
+
+        if 'id' in goto_rule['when']:
+            answer_index = goto_rule['when']['id']
+            filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
+            if len(filtered) == 1:
+                return evaluate_rule(goto_rule, filtered[0]['value'])
+
+        elif 'meta' in goto_rule['when']:
+            return evaluate_metadata_rule(goto_rule, metadata)
```

Comment:
I think it would be more easily understood if this matched what we do for the id rule above - a method to get the value from the metadata, and then use the `evaluate_rule`, something like:
```
key = goto_rule['when']['meta']
value = get_metadata_value(metadata, key)
return evaluate_rule(goto_rule, value)
```